### ISSUE TITLE
google-cloud-sdk: Move python2 to optdepends

### ISSUE THREAD

**Issue by lubo** [sudoforge/pkgbuilds#17]
Since only a couple of utilities in the SDK still require Python 2, the users not using them shouldn't be required to install it.

**Comment by sudoforge** [comment]
I'm torn on this, because I don't use either `dev_appserver.py` or `endpointscfg` and have no need for `python2` on my system, but it **is** required in order to run those two utilities, and it **is** still looked for by the `gcloud` sdk by default:

[`gcloud topic startup`][0]:
> The gcloud CLI runs under Python. On Windows, you have the option of installing a bundled Python interpreter, but on Mac and Linux you must have a Python interpreter available somewhere on your system. gcloud will attempt to locate an interpreter on your system PATH by looking for the following binaries (in this order):
>   * python2
>   * python27
>   * python
>   * python3

[0]: https://cloud.google.com/sdk/gcloud/reference/topic/startup

The interpreter used by the SDK is set via the [`CLOUDSDK_PYTHON` variable][1], so users of this package are not using `python2` by default, instead, the SDK is using the `python` interpreter (which is `python3`).

**Can I get a little more context into why this change would be helpful for you? Is it a space concern?**

To note, `python2` is a required dependency of several other packages -- 31 to be exact (not including `python2-*` subpackages or `google-cloud-sdk`):

```
➜ pacman -Sii python2 | grep '^Required By' | sed -e 's/.*:\s*//' -e 's/  /\r\n/g' | grep -v '^python2-' | grep -v '^google-cloud-sdk'
asciidoc
dblatex
epydoc
getmail
gmock
inkscape
ipython2
java-openjfx
java11-openjfx
java8-openjfx
kodi-bin
kodi-gbm
kodi-wayland
libgda
mailman
mercurial
mysql-python
mysql-workbench
ntop
pychecker
pydb
pylibacl
pyrit
rdiff-backup
scribus
singularity
tellico
texmacs
tuxpaint
wesnoth
wifite
```

[1]: https://github.com/sudoforge/pkgbuilds/blob/master/google-cloud-sdk/google-cloud-sdk.sh#L2

**Comment by lubo** [comment]
>I'm torn on this, because I don't use either dev_appserver.py or endpointscfg and have no need for python2 on my system, but it is required in order to run those two utilities, and it is still looked for by the gcloud sdk by default:

It's normal for certain functionality to require additional dependencies and it's normal for the functionality to not work when the requirements aren't met. Also, the fact it's still looked for by `gcloud` is IMO another argument for not requiring Python 2 by default. Currently, the chances that Python 2 will be used due to misconfiguration are higher than if it wasn't installed by default. Respectfully, I don't understand what you're torn about.

>Can I get a little more context into why this change would be helpful for you? Is it a space concern?

I simply don't wanna be forced to have potentially insecure software installed on my system and risking it being used unbeknown to me, especially if I've no legitimate use for it. `google-cloud-sdk` is the last package on my system requiring it.

**Comment by sudoforge** [comment]
I'm sold -- patch incoming shortly. Thanks for opening a discussion about this!

**Comment by sudoforge** [comment]
Closed in `b47d0b3e03c2f02d8f6c6ab1e2f9a2d52d3d6dce` ;  I forgot to add an issue reference. `285.0.1-2` is up in the AUR!
